packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/advanced-duration.tsx

```diff
@@ -58,7 +96,11 @@ const AdvancedDuration = observer(
 
         const has_error = !!validation_errors?.duration?.length;
 
-        const { name_plural, name } = getUnitMap()[advanced_duration_unit];
+        const { name_plural, name } = getUnitMap()[advanced_duration_unit as keyof ReturnType<typeof getUnitMap>] as {
+            name_plural?: string;
+            name_singular?: string;
+            name?: string;
+        };
```

Comment:
`getUnitMap()` returns an object:
```
{
        d: { name_plural: localize('days'), name_singular: localize('day') },
        h: { name_plural: localize('hours'), name_singular: localize('hour') },
        m: { name_plural: localize('minutes'), name_singular: localize('minute') },
        s: { name: localize('seconds') },
        t: { name_plural: localize('ticks'), name_singular: localize('tick') },
    };
```
The problem is that `name_plural` and `name` do not exist in each field, so TS was giving a lot of errors. That is why I added `as {name_plural?:string and etc}`, but it's only for satisfying TS. Please, if you have good suggestions for this case, share with me.

Reply:
@kate-deriv please declare a type for the object returned from getUnitMap and each object type inside it
```
type TUnitMap = {
            name_plural?: string;
            name_singular?: string;
            name?: string;
        }
```
no need for this huge assertion:
```
as {
            name_plural?: string;
            name_singular?: string;
            name?: string;
        };
```